### PR TITLE
chore(STRK-167): address PR #1233 review comments

### PR DESCRIPTION
**Pull request by lbruton**
Follow-up to [#1233](https://github.com/lbruton/StakTrakr/pull/1233) — addresses the 4 advisory review threads (Copilot + Codacy) that posted ~1.5 min after that PR merged.

| # | Finding | Fix |
|---|---------|-----|
| Copilot `diff-engine.js:330` | `_instanceKey` didn't trim `numistaId` (`" 12345 "` ≠ `"12345"`; `sanitizeObjectFields` preserves spaces) | Trim `numistaId` in `_instanceKey` **and** the `changeLog.js` fallback (keeps AC-2 equivalence); added a regression test |
| Copilot `diff-engine.js:380` | `collapseByInstanceKey` doc claimed all rows "shallow-cloned" but non-numistaId rows pass by reference | Tightened the doc to match the implementation |
| Copilot `index.html:7275` | `.sr-only` re-defined inline (with a needless `!important`) when it's already global in `css/styles.css` | Dropped the inline copy; badge relies on the global rule |
| Codacy `changeLog.js:44` | `_fallbackItemKey` is a shadow of `_instanceKey` | Kept (intentional DiffEngine-absent safety net) + added a `duplication-ok` marker and clarifying comment |

## Tests
- [x] `npm run test:unit` — 217 pass (new numistaId-trim assertion in `diff-engine-instance-key.test.js`)
- [x] `numista-import-onboarding.spec.js` — 7 pass (badge sr-only text still resolves via the global rule)

No version bump (chore — review-comment cleanup, no user-facing behavior change beyond the trim edge case).

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed issue where whitespace in ID and year fields caused key matching failures, preventing duplicate items from being created.

* **Tests**
  * Added unit test verifying whitespace trimming behavior for key generation.

* **Refactor**
  * Consolidated CSS accessibility utilities to use global styles and improved documentation clarity.

### CODE REVIEW

File: js/changeLog.js [hunk 1]
```diff
@@ -37,8 +39,9 @@ const _fallbackItemKey = (item) => {
       String(v == null ? "" : v)
         .trim()
         .toLowerCase();
+    const nid = String(item.numistaId == null ? "" : item.numistaId).trim();
     const year = item.year == null ? "" : String(item.year);
-    return `${item.numistaId}|${year}|${norm(item.grade)}|${norm(item.certNumber)}`;
+    return `${nid}|${year}|${norm(item.grade)}|${norm(item.certNumber)}`;
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/changeLog.js:44]: Fixed in 54ddcf3b — the `year` segment is now trimmed in both `DiffEngine._instanceKey` and the `changeLog.js` fallback, matching the numistaId trim. Added a regression assertion.

File: js/diff-engine.js [hunk 2]
```diff
@@ -325,8 +326,9 @@ const DiffEngine = {
       String(v == null ? "" : v)
         .trim()
         .toLowerCase();
+    const nid = String(item.numistaId == null ? "" : item.numistaId).trim();
     const year = item.year == null ? "" : String(item.year);
-    return `${item.numistaId}|${year}|${norm(item.grade)}|${norm(item.certNumber)}`;
+    return `${nid}|${year}|${norm(item.grade)}|${norm(item.certNumber)}`;
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/diff-engine.js:331]: Fixed in 54ddcf3b — the `year` segment is now trimmed in both `DiffEngine._instanceKey` and the `changeLog.js` fallback, matching the numistaId trim. Added a regression assertion.
